test/modellib/ModelLib.cpp

```diff
@@ -99,6 +102,28 @@ void ModelLibBuilder::setRandomNumberGeneratorSeed(const std::string &envVar) {
   }
 }
 
+#ifndef _WIN32
+bool ModelLibBuilder::checkSharedLibInstruction(
+    std::string instructionName, std::string sharedLibName) {
+  if (instructionName.empty())
+    return true;
+  void *handle;
+  handle = dlopen(sharedLibName.c_str(), RTLD_LAZY);
+  if (handle == NULL) {
+    printf("%s\n", dlerror());
+    return false;
+  }
+  int *dptr;
+  dptr = (int *)dlsym(handle, instructionName.c_str());
+  if (dptr == NULL) {
+    printf("%s\n", dlerror());
+    return false;
```

Comment:
Should call dlclose before returning.

Reply:
@gongsu832 Thanks. I remove ifdef _WIN32 in https://github.com/onnx/onnx-mlir/pull/1335/commits/e20853f1e8c86cab47539264e4004990df373f0c, but it seems it didn't work for WIN32. Is something missing?

I also test another way for WIN32, but it still fails.

Reply:
It's failing to compile `ModelLib.cpp`
https://dev.azure.com/onnx-pipelines/d175d065-ea6c-4983-99d3-ae963744886b/_apis/build/builds/24898/logs/14

Reply:
If you look at the code in ExecutionSession builder, it seems that this can be entirely avoided using the llvm code
```
  _sharedLibraryHandle =
      llvm::sys::DynamicLibrary::getPermanentLibrary(sharedLibPath.c_str());
  if (!_sharedLibraryHandle.isValid())
    throw std::runtime_error(reportLibraryOpeningError(sharedLibPath));
```

There is no special casing whatever for Window or Linux... So it seems a more robust solution as they already deal with it in LLVM.